rpc/wrtc_server_stream.go

```diff
@@ -269,30 +269,24 @@ func (s *webrtcServerStream) processHeaders(headers *webrtcpb.RequestHeaders) {
 	}
 
 	s.ch.server.counters.HeadersProcessed.Add(1)
-	s.ch.server.processHeadersMu.RLock()
-	s.ch.server.processHeadersWorkers.Add(1)
-	s.ch.server.processHeadersMu.RUnlock()
 
 	// Check if context has errored: underlying server may have been `Stop`ped,
-	// in which case we mark this processHeaders worker as `Done` and return.
-	if err := s.ch.server.ctx.Err(); err != nil {
-		s.ch.server.processHeadersWorkers.Done()
+	// in which case we return.
+	if err := s.ch.server.processHeadersWorkers.Context().Err(); err != nil {
 		return
 	}
 
 	// take a ticket
 	select {
 	case s.ch.server.callTickets <- struct{}{}:
 	default:
-		s.ch.server.processHeadersWorkers.Done()
 		s.closeWithSendError(status.Error(codes.ResourceExhausted, "too many in-flight requests"))
 		return
 	}
 
 	s.headersReceived = true
 	utils.PanicCapturingGo(func() {
```

Comment:
I think this should probably be a `s.ch.server.processHeadersWorkers.Add(` call instead of a `PanicCapturingGo` call.